### DISCUSSION TITLE
Styles not picked up in dev (with react-bootstrap and styled-components)

### DISCUSSION THREAD

**Discussion by FBosler** [vercel/next.js#13923] [Help] [answered]
# Bug report

In `yarn dev` styles from react-bootstrap are not picked up on the root page. In the example at had leading to the "Join the movement"-button 
![image](https://user-images.githubusercontent.com/14016698/84116704-9e58e700-aa30-11ea-9b57-54b371002f44.png)

not picking up the styles from 
```javascript
export const VerticalCenterWrapper = styled("div")`
    height: 100% !important;
    display: flex;
`;
```

## To Reproduce

1. Go to https://github.com/FBosler/next-strange-ssr-formatting
2. Run `yarn` or `npm install` locally
3. Run `yarn dev`
4. See how button does not pick up the [style](https://github.com/FBosler/next-strange-ssr-formatting/blob/eaa72f050b9f60127fabd1ca84349887dedd03f9/components/Landing/styles.js#L64)

## Expected behavior

Would expect all styles to be applied or see some form of error

## Screenshots
This is how the page looks in `dev`
![image](https://user-images.githubusercontent.com/14016698/84117204-6ef6aa00-aa31-11ea-90cc-aca0f3fb259b.png)

This is how the page looks after building (and is supposed to look)
![image](https://user-images.githubusercontent.com/14016698/84117138-55556280-aa31-11ea-86b8-740bd30f3f1e.png)

## System information

- OS: macOS Catalina (10.15.4)
- Browser: Chrome
- Version of Next.js: 9.4.2
- Version of Node.js: 13.13.0

**Comment by mattcarlotta** [comment, marked as answer]
The problem is that your `LandingBackground` contains invalid CSS syntax:

```js
export const LandingBackground = styled("div")`
    background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url("${(
      props
    ) => props.url}");
    background-position: 50% 60%;
    background-size: cover;
    border-bottom: grey 1px solid;
    height: 200px;
    ${device.laptop} { 
        height: 300px;
      }
`;
```

Not sure what this code is intended to do, but it's preventing styled-components from traversing children and applying styles to them:
```js
    ${device.laptop} { 
        height: 300px;
      }
```

Instead, it's applying an object of CSS to the `LandingBackground`:
![](https://i.imgur.com/Mp6t3za.png)

Removing the above code works as expected:
![](https://i.imgur.com/WHzkEYI.png)
![](https://i.imgur.com/9Fbd8bu.png)

**Reply by FBosler** [reply]
Amazing, thx so much. the `device.laptop` includes a mediaquery. Unfortunately I had renamed the laptop to match the bootstrap logic.

Any idea, why it's not failing in production?
